ui/components/institutional/note-to-trader/note-to-trader.tsx

```diff
@@ -1,28 +1,54 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import {
+  BackgroundColor,
+  BlockSize,
+  BorderRadius,
   Display,
   FlexDirection,
   JustifyContent,
 } from '../../../helpers/constants/design-system';
 import { Label, Box, Text } from '../../component-library';
+import { Textarea } from '../../component-library/textarea';
+import { setNoteToTraderMessage } from '../../../store/institutional/institution-background';
+import {
+  getIsNoteToTraderSupported,
+  State,
+} from '../../../selectors/institutional/selectors';
+import { toChecksumHexAddress } from '../../../../shared/modules/hexstring-utils';
+import { useConfirmContext } from '../../../pages/confirmations/context/confirm';
+import { getConfirmationSender } from '../../../pages/confirmations/components/confirm/utils';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { isSignatureTransactionType } from '../../../pages/confirmations/utils';
 
-type NoteToTraderProps = {
-  placeholder: string;
-  maxLength: number;
-  onChange: (value: string) => void;
-  noteText: string;
-  labelText: string;
-};
+const NoteToTrader: React.FC = () => {
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+  const [noteText, setNoteText] = useState('');
+
+  const { currentConfirmation } = useConfirmContext();
+  const isSignature = isSignatureTransactionType(currentConfirmation);
+  const { from } = getConfirmationSender(currentConfirmation);
+  const fromChecksumHexAddress = toChecksumHexAddress(from || '');
+  const isNoteToTraderSupported = useSelector((state: State) =>
+    getIsNoteToTraderSupported(state, fromChecksumHexAddress),
+  );
 
-const NoteToTrader: React.FC<NoteToTraderProps> = ({
-  placeholder,
-  maxLength,
-  onChange,
-  noteText,
-  labelText,
-}) => {
-  return (
-    <Box className="confirm-page-container-content__data">
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      dispatch(setNoteToTraderMessage(noteText));
+    }, 700);
+
+    return () => clearTimeout(timer);
+  }, [noteText]);
+
+  return isNoteToTraderSupported && !isSignature ? (
```

Comment:
[nit]: Can instead be expressed with an early return which is arguably more readable:

```tsx
if (!isNotToTraderSupported || isSignature) {
  return null
}
```